code/datums/components/_component.dm

```diff
@@ -137,6 +137,10 @@
 	return
 
 /datum/component/proc/_GetInverseTypeList(our_type = type)
+#if DM_VERSION > 511
+#warning Remove this hack for http://www.byond.com/forum/?post=73469

```

Comment:
or just:
```
if(!our_type)
    our_type = type
```
No need for nonsense (and, imo, an ugly use of default arguments)

Reply:
Especially since, as lummox reveals in the bug report, default args just generate the if anyway!

Reply:
The bug is actually the var/current_type = parent_type line below this. src is null because of some fuckery

Reply:
and its caused BY the default arg.

Reply:
yeah this workaround was discussed in the thread and is far more elegant than what's been done here